src/cmd/exec.rs

```diff
@@ -70,9 +72,23 @@ pub fn exec(
         return Err(anyhow!("No context matching {}", context_name));
     }
 
+    let print_context = if override_context_headers {
+        true
+    } else if override_no_context_headers {
+        false
+    } else {
+        settings.behavior.print_context_in_exec.should_print_headers()
+    };
```

Comment:
I'm not entirely happy with this if statement... give it a think and see if I'm missing something here? It works, it just feels gross. I dunno, maybe I'm overanalyzing it. 🤷